controller/kubernetes_controller.go

```diff
@@ -387,3 +427,52 @@ func (kc *KubernetesController) detectWorkload(p *v1.Pod, ks *types.KubernetesSt
 	}
 	return "", ""
 }
+
+func (kc *KubernetesController) cleanupVolumeAttachment(volume *longhorn.Volume, ks *types.KubernetesStatus) {
+	// PV and PVC should exist. Pod should be Pending
+	if !(ks.PVStatus == string(v1.VolumeBound) && ks.PVCName != "" && ks.LastPVCRefAt == "") {
+		return
+	}
+	if !(ks.PodName != "" && ks.PodStatus == string(v1.PodPending) && ks.LastPodRefAt == "") {
+		return
+	}
+
+	va, err := kc.getVolumeAttachment(ks)
+	if err != nil {
+		logrus.Errorf("failed to get VolumeAttachment in cleanupVolumeAttachment: %v", err)
+		return
+	}
+	if va == nil {
+		return
+	}
+
+	cleanupFlag, err := kc.ds.IsNodeDownOrDeleted(va.Spec.NodeName)
+	if err != nil {
+		logrus.Errorf("failed to detect node %v in cleanupVolumeAttachment: %v", va.Spec.NodeName, err)
+		return
+	}
+
+	// the node VolumeAttachment on is declared `NotReady` or doesn't exist.
+	if cleanupFlag {
+		err := kc.kubeClient.StorageV1beta1().VolumeAttachments().Delete(va.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			logrus.Errorf("failed to delete VolumeAttachment %v in cleanupVolumeAttachment: %v", va.Name, err)
+			return
+		}
+	}
+	kc.eventRecorder.Eventf(volume, v1.EventTypeNormal, EventReasonDelete, "Cleanup Volume Attachment on 'NotReady' Node: %v", va.Name)
```

Comment:
Should this in the `if` clause above? Otherwise, this event will be sent every time when the node is not down.